src/globals.c

```diff
@@ -212,13 +213,37 @@ bool set_format_string(char *userinput)
 	return true;
 }
 
+bool set_state_dir(char *dir)
+{
+	if (dir == NULL || dir[0] == '\0') {
+		return false;
+	}
+
+	/* TODO: more validation of input? */
+	if (dir[0] != '/') {
+		printf("statedir must be a full path starting with '/', not '%c'\n", dir[0]);
+		return false;
+	}
+
+	if (state_dir) {
+		free(state_dir);
+	}
+	string_or_die(&state_dir, "%s", dir);
+
+	return true;
+}
+
 bool init_globals(void)
 {
 	struct stat statbuf;
 	int ret;
 
 	gettimeofday(&start_time, NULL);
 
+	if (state_dir == NULL) {
+		string_or_die(&state_dir, "%s", STATE_DIR);
+	}
+
 	/* insure path_prefix is absolute, at least '/', ends in '/',
 	 * and is a valid dir */
 	if (path_prefix != NULL) {
```

Comment:
Latest code in master sets path_prefix in the set_path_prefix() function, so there are likely merge conflicts here. Can you rebase this patch?